Header header
bool if_good
